tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,339 @@
+			console.log(`Eth link is: ${ethLink}`)
+			return ethLink === '1';
+		})
+
+		// SSH access shouldn't be possible - as the device shouldn't be booting?
+		// the CM4 takes a while to get into a state where we can SSH into it anyway after powering on - so we want to try a good number of times.
+		try {
+			await this.worker.executeCommandInHostOS('echo -n pass',this.link, { max_tries: 20, interval: 1000 });
+			// if we manage to SSH into the DUT, then something has gone wrong - so throw an error which will fail the test
+			throw new Error(`DUT was still reachable over SSH`)
+		} catch (e){
+			// we want / are expecting a failure here, so if the SSH fails after the specified number of retries, we want to return
+			console.log(`DUT was not reachable over SSH`)
+			return
+		}
+	}
+
+	async testBootloaderIntegrity() {
+		/* Modify boot.sig to fail authentication */
+		await this.worker.executeCommandInHostOS(
+			"sed -i -e '1s/^.\\(.*\\)$/\\1/' /mnt/rpi/boot.sig",
+			this.link
+		)
+		await this.worker.executeCommandInHostOS('reboot', this.link)
+	    await this.test.resolves(
+			this.waitForFailedBoot(),
+			'Bootloader will not load configuration that fails signature verification',
+		)
+		await this.resetDUT();
+	}
+
+	async testBootloaderConfigIntegrity() {
+		/* Modify config.txt inside boot.img */
+		await this.worker.executeCommandInHostOS(
+			"tmpDir=$(mktemp -d) && mount -o loop /mnt/rpi/boot.img ${tmpDir} && sed -i 's/$/#comment/' ${tmpDir}/config.txt && umount ${tmpDir} && rm -rf ${tmpDir}",
+			this.link
+		)
+		await this.worker.executeCommandInHostOS('reboot', this.link)
+		await this.test.resolves(
+			this.waitForFailedBoot(),
+			'Bootloader will not load configuration that fails signature verification',
+		)
+		await this.resetDUT();
+	}
+}
+
+class testSecureBoot {
+	constructor(impl) {
+		this.impl = impl;
+	}
+
+	async run(test) {
+		if ( await this.impl.isSecureBootSupported() ) {
+			if ( await this.impl.isSecureBootEnabled() ) {
+				await this.impl.testFullDiskEncryption();
+				await this.impl.testModuleVerification();
+				await this.impl.testBootloaderIntegrity();
+				await this.impl.testBootloaderConfigIntegrity();
+			} else {
+				test.comment('Secure boot is not enabled');
+			}
+		} else {
+			test.comment('Secure boot is not supported by firmware');
+		}
+	}
+}
+
+module.exports = {
+	title: 'secure boot tests',
+	tests: [
+		{
+			title: 'check UEFI secure boot',
+			run: async function(test) {
+				const impl = new testSecureBoot(new uefiSecureBoot(test,
+					this.worker,
+					this.suite, this.os.image.path,
+					{"name": "pcan_netdev", "headersVersion": "2.108.6"}));
+				await impl.run(test);
+			},
+		},
+		{
+			title: 'check RPI secure boot',
+			run: async function(test) {
+				const impl = new testSecureBoot(new rpiSecureBoot(test,
+					this.worker,
+					this.suite, this.os.image.path,
+					{"name": "", "headersVersion": "4.0.16"}));
```

Comment:
this is based on the latest non-sb cm4 image isn't it? 

Is this something thats fine to leave long term? Or do we always want it to be from latest prod for example?

Reply:
Once there is a version deployed in production for the `raspberrypicm4-ioboard-sb` we will replace with that headers version.

Tecnically it doesn't need to be the latest headers as they are `clean` kernel headers. Also, the hello module we are building is so simple that kernel API changes in headers won't really affect it. So, adding extra code to use latest would be over-engineering for this use case.